// app-store-server-api/dist/esm/Models.js
/**
 * https://developer.apple.com/documentation/appstoreserverapi/environment
 */
export var Environment;
(function (Environment) {
    Environment["Production"] = "Production";
    Environment["Sandbox"] = "Sandbox";
})(Environment || (Environment = {}));
export var SortParameter;
(function (SortParameter) {
    SortParameter["Ascending"] = "ASCENDING";
    SortParameter["Descending"] = "DESCENDING";
})(SortParameter || (SortParameter = {}));
export var ProductTypeParameter;
(function (ProductTypeParameter) {
    ProductTypeParameter["AutoRenewable"] = "AUTO_RENEWABLE";
    ProductTypeParameter["NonRenewable"] = "NON_RENEWABLE";
    ProductTypeParameter["Consumable"] = "CONSUMABLE";
    ProductTypeParameter["NonConsumable"] = "NON_CONSUMABLE";
})(ProductTypeParameter || (ProductTypeParameter = {}));
export var TransactionHistoryVersion;
(function (TransactionHistoryVersion) {
    /** @deprecated */
    TransactionHistoryVersion["v1"] = "v1";
    TransactionHistoryVersion["v2"] = "v2";
})(TransactionHistoryVersion || (TransactionHistoryVersion = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/offerdiscounttype
 */
export var OfferDiscountType;
(function (OfferDiscountType) {
    OfferDiscountType["FreeTrial"] = "FREE_TRIAL";
    OfferDiscountType["PayAsYouGo"] = "PAY_AS_YOU_GO";
    OfferDiscountType["PayUpFront"] = "PAY_UP_FRONT";
})(OfferDiscountType || (OfferDiscountType = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/inappownershiptype
 */
export var OwnershipType;
(function (OwnershipType) {
    OwnershipType["Purchased"] = "PURCHASED";
    OwnershipType["FamilyShared"] = "FAMILY_SHARED";
})(OwnershipType || (OwnershipType = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/type
 */
export var TransactionType;
(function (TransactionType) {
    TransactionType["AutoRenewableSubscription"] = "Auto-Renewable Subscription";
    TransactionType["NonConsumable"] = "Non-Consumable";
    TransactionType["Consumable"] = "Consumable";
    TransactionType["NonRenewingSubscription"] = "Non-Renewing Subscription";
})(TransactionType || (TransactionType = {}));
/**
 * https://developer.apple.com/documentation/appstoreservernotifications/transactionreason
 */
export var TransactionReason;
(function (TransactionReason) {
    TransactionReason["Purchase"] = "PURCHASE";
    TransactionReason["Renewal"] = "RENEWAL";
})(TransactionReason || (TransactionReason = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/status
 */
export var SubscriptionStatus;
(function (SubscriptionStatus) {
    SubscriptionStatus[SubscriptionStatus["Active"] = 1] = "Active";
    SubscriptionStatus[SubscriptionStatus["Expired"] = 2] = "Expired";
    SubscriptionStatus[SubscriptionStatus["InBillingRetry"] = 3] = "InBillingRetry";
    SubscriptionStatus[SubscriptionStatus["InBillingGracePeriod"] = 4] = "InBillingGracePeriod";
    SubscriptionStatus[SubscriptionStatus["Revoked"] = 5] = "Revoked";
})(SubscriptionStatus || (SubscriptionStatus = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/autorenewstatus
 */
export var AutoRenewStatus;
(function (AutoRenewStatus) {
    AutoRenewStatus[AutoRenewStatus["Off"] = 0] = "Off";
    AutoRenewStatus[AutoRenewStatus["On"] = 1] = "On";
})(AutoRenewStatus || (AutoRenewStatus = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/expirationintent
 */
export var ExpirationIntent;
(function (ExpirationIntent) {
    ExpirationIntent[ExpirationIntent["Canceled"] = 1] = "Canceled";
    ExpirationIntent[ExpirationIntent["BillingError"] = 2] = "BillingError";
    ExpirationIntent[ExpirationIntent["RejectedPriceIncrease"] = 3] = "RejectedPriceIncrease";
    ExpirationIntent[ExpirationIntent["ProductUnavailable"] = 4] = "ProductUnavailable";
})(ExpirationIntent || (ExpirationIntent = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/offertype
 */
export var OfferType;
(function (OfferType) {
    OfferType[OfferType["Introductory"] = 1] = "Introductory";
    OfferType[OfferType["Promotional"] = 2] = "Promotional";
    OfferType[OfferType["SubscriptionOfferCode"] = 3] = "SubscriptionOfferCode";
    OfferType[OfferType["WinBackOffer"] = 4] = "WinBackOffer";
})(OfferType || (OfferType = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/priceincreasestatus
 */
export var PriceIncreaseStatus;
(function (PriceIncreaseStatus) {
    PriceIncreaseStatus[PriceIncreaseStatus["NoResponse"] = 0] = "NoResponse";
    PriceIncreaseStatus[PriceIncreaseStatus["Consented"] = 1] = "Consented";
})(PriceIncreaseStatus || (PriceIncreaseStatus = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/orderlookupstatus
 */
export var OrderLookupStatus;
(function (OrderLookupStatus) {
    OrderLookupStatus[OrderLookupStatus["Valid"] = 0] = "Valid";
    OrderLookupStatus[OrderLookupStatus["Invalid"] = 1] = "Invalid";
})(OrderLookupStatus || (OrderLookupStatus = {}));
export function isDecodedNotificationDataPayload(decodedNotificationPayload) {
    return "data" in decodedNotificationPayload;
}
export function isDecodedNotificationSummaryPayload(decodedNotificationPayload) {
    return "summary" in decodedNotificationPayload;
}
/**
 * https://developer.apple.com/documentation/appstoreservernotifications/notificationtype
 */
export var NotificationType;
(function (NotificationType) {
    NotificationType["ConsumptionRequest"] = "CONSUMPTION_REQUEST";
    NotificationType["DidChangeRenewalPref"] = "DID_CHANGE_RENEWAL_PREF";
    NotificationType["DidChangeRenewalStatus"] = "DID_CHANGE_RENEWAL_STATUS";
    NotificationType["DidFailToRenew"] = "DID_FAIL_TO_RENEW";
    NotificationType["DidRenew"] = "DID_RENEW";
    NotificationType["Expired"] = "EXPIRED";
    NotificationType["ExternalPurchaseToken"] = "EXTERNAL_PURCHASE_TOKEN";
    NotificationType["GracePeriodExpired"] = "GRACE_PERIOD_EXPIRED";
    NotificationType["OfferRedeemed"] = "OFFER_REDEEMED";
    NotificationType["OneTimeCharge"] = "ONE_TIME_CHARGE";
    NotificationType["PriceIncrease"] = "PRICE_INCREASE";
    NotificationType["Refund"] = "REFUND";
    NotificationType["RefundDeclined"] = "REFUND_DECLINED";
    NotificationType["RefundReversed"] = "REFUND_REVERSED";
    NotificationType["RenewalExtended"] = "RENEWAL_EXTENDED";
    NotificationType["RenewalExtension"] = "RENEWAL_EXTENSION";
    NotificationType["Revoke"] = "REVOKE";
    NotificationType["Subscribed"] = "SUBSCRIBED";
    NotificationType["Test"] = "TEST";
})(NotificationType || (NotificationType = {}));
/**
 * https://developer.apple.com/documentation/appstoreservernotifications/subtype
 */
export var NotificationSubtype;
(function (NotificationSubtype) {
    NotificationSubtype["InitialBuy"] = "INITIAL_BUY";
    NotificationSubtype["Resubscribe"] = "RESUBSCRIBE";
    NotificationSubtype["Downgrade"] = "DOWNGRADE";
    NotificationSubtype["Upgrade"] = "UPGRADE";
    NotificationSubtype["AutoRenewEnabled"] = "AUTO_RENEW_ENABLED";
    NotificationSubtype["AutoRenewDisabled"] = "AUTO_RENEW_DISABLED";
    NotificationSubtype["Voluntary"] = "VOLUNTARY";
    NotificationSubtype["BillingRetry"] = "BILLING_RETRY";
    NotificationSubtype["PriceIncrease"] = "PRICE_INCREASE";
    NotificationSubtype["ProductNotForSale"] = "PRODUCT_NOT_FOR_SALE";
    NotificationSubtype["GracePeriod"] = "GRACE_PERIOD";
    NotificationSubtype["BillingRecovery"] = "BILLING_RECOVERY";
    NotificationSubtype["Pending"] = "PENDING";
    NotificationSubtype["Accepted"] = "ACCEPTED";
    NotificationSubtype["Summary"] = "SUMMARY";
    NotificationSubtype["Failure"] = "FAILURE";
})(NotificationSubtype || (NotificationSubtype = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/sendattemptresult
 */
export var SendAttemptResult;
(function (SendAttemptResult) {
    SendAttemptResult["Success"] = "SUCCESS";
    SendAttemptResult["TimedOut"] = "TIMED_OUT";
    SendAttemptResult["TlsIssue"] = "TLS_ISSUE";
    SendAttemptResult["CircularRedirect"] = "CIRCULAR_REDIRECT";
    SendAttemptResult["NoResponse"] = "NO_RESPONSE";
    SendAttemptResult["SocketIssue"] = "SOCKET_ISSUE";
    SendAttemptResult["UnsupportedCharset"] = "UNSUPPORTED_CHARSET";
    SendAttemptResult["InvalidResponse"] = "INVALID_RESPONSE";
    SendAttemptResult["PrematureClose"] = "PREMATURE_CLOSE";
    SendAttemptResult["Other"] = "OTHER";
})(SendAttemptResult || (SendAttemptResult = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/extendreasoncode
 */
export var ExtendReasonCode;
(function (ExtendReasonCode) {
    ExtendReasonCode[ExtendReasonCode["UNDECLARED"] = 0] = "UNDECLARED";
    ExtendReasonCode[ExtendReasonCode["CUSTOMER_SATISFACTION"] = 1] = "CUSTOMER_SATISFACTION";
    ExtendReasonCode[ExtendReasonCode["OTHER_REASON"] = 2] = "OTHER_REASON";
    ExtendReasonCode[ExtendReasonCode["SERVICE_ISSUE"] = 3] = "SERVICE_ISSUE";
})(ExtendReasonCode || (ExtendReasonCode = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/accounttenure
 */
export var AccountTenure;
(function (AccountTenure) {
    AccountTenure[AccountTenure["Undeclared"] = 0] = "Undeclared";
    AccountTenure[AccountTenure["Between0_3Days"] = 1] = "Between0_3Days";
    AccountTenure[AccountTenure["Between3_10Days"] = 2] = "Between3_10Days";
    AccountTenure[AccountTenure["Between10_30Days"] = 3] = "Between10_30Days";
    AccountTenure[AccountTenure["Between30_90Days"] = 4] = "Between30_90Days";
    AccountTenure[AccountTenure["Between90_180Days"] = 5] = "Between90_180Days";
    AccountTenure[AccountTenure["Between180_365Days"] = 6] = "Between180_365Days";
    AccountTenure[AccountTenure["Over365Days"] = 7] = "Over365Days";
})(AccountTenure || (AccountTenure = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/consumptionstatus
 */
export var ConsumptionStatus;
(function (ConsumptionStatus) {
    ConsumptionStatus[ConsumptionStatus["Undeclared"] = 0] = "Undeclared";
    ConsumptionStatus[ConsumptionStatus["NotConsumed"] = 1] = "NotConsumed";
    ConsumptionStatus[ConsumptionStatus["PartiallyConsumed"] = 2] = "PartiallyConsumed";
    ConsumptionStatus[ConsumptionStatus["FullyConsumed"] = 3] = "FullyConsumed";
})(ConsumptionStatus || (ConsumptionStatus = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/deliverystatus
 */
export var DeliveryStatus;
(function (DeliveryStatus) {
    DeliveryStatus[DeliveryStatus["DeliveredAndWorkingProperly"] = 0] = "DeliveredAndWorkingProperly";
    DeliveryStatus[DeliveryStatus["DidntDeliverDueToQualityIssue"] = 1] = "DidntDeliverDueToQualityIssue";
    DeliveryStatus[DeliveryStatus["DeliveredWrongItem"] = 2] = "DeliveredWrongItem";
    DeliveryStatus[DeliveryStatus["DidntDeliverDueToServerOutage"] = 3] = "DidntDeliverDueToServerOutage";
    DeliveryStatus[DeliveryStatus["DidntDeliverDueToInGameCurrencyChange"] = 4] = "DidntDeliverDueToInGameCurrencyChange";
    DeliveryStatus[DeliveryStatus["DidntDeliverDueToOtherReason"] = 5] = "DidntDeliverDueToOtherReason";
})(DeliveryStatus || (DeliveryStatus = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/lifetimedollarspurchased
 */
export var LifetimeDollarsPurchased;
(function (LifetimeDollarsPurchased) {
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Undeclared"] = 0] = "Undeclared";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["ZeroUSD"] = 1] = "ZeroUSD";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Between_0_01_And_49_99USD"] = 2] = "Between_0_01_And_49_99USD";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Between_50_And_99_99USD"] = 3] = "Between_50_And_99_99USD";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Between_100_And_499_99USD"] = 4] = "Between_100_And_499_99USD";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Between_500_And_999_99USD"] = 5] = "Between_500_And_999_99USD";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Between_1000_And_1999_99USD"] = 6] = "Between_1000_And_1999_99USD";
    LifetimeDollarsPurchased[LifetimeDollarsPurchased["Over_2000USD"] = 7] = "Over_2000USD";
})(LifetimeDollarsPurchased || (LifetimeDollarsPurchased = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/lifetimedollarsrefunded
 */
export var LifetimeDollarsRefunded;
(function (LifetimeDollarsRefunded) {
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Undeclared"] = 0] = "Undeclared";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["ZeroUSD"] = 1] = "ZeroUSD";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Between_0_01_And_49_99USD"] = 2] = "Between_0_01_And_49_99USD";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Between_50_And_99_99USD"] = 3] = "Between_50_And_99_99USD";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Between_100_And_499_99USD"] = 4] = "Between_100_And_499_99USD";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Between_500_And_999_99USD"] = 5] = "Between_500_And_999_99USD";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Between_1000_And_1999_99USD"] = 6] = "Between_1000_And_1999_99USD";
    LifetimeDollarsRefunded[LifetimeDollarsRefunded["Over_2000USD"] = 7] = "Over_2000USD";
})(LifetimeDollarsRefunded || (LifetimeDollarsRefunded = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/platform
 */
export var Platform;
(function (Platform) {
    Platform[Platform["Undeclared"] = 0] = "Undeclared";
    Platform[Platform["Apple"] = 1] = "Apple";
    Platform[Platform["NonApple"] = 2] = "NonApple";
})(Platform || (Platform = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/playtime
 */
export var PlayTime;
(function (PlayTime) {
    PlayTime[PlayTime["Undeclared"] = 0] = "Undeclared";
    PlayTime[PlayTime["Between0_5Minutes"] = 1] = "Between0_5Minutes";
    PlayTime[PlayTime["Between5_60Minutes"] = 2] = "Between5_60Minutes";
    PlayTime[PlayTime["Between1_6Hours"] = 3] = "Between1_6Hours";
    PlayTime[PlayTime["Between6_24Hours"] = 4] = "Between6_24Hours";
    PlayTime[PlayTime["Between1_4Days"] = 5] = "Between1_4Days";
    PlayTime[PlayTime["Between4_16Days"] = 6] = "Between4_16Days";
    PlayTime[PlayTime["Over16Days"] = 7] = "Over16Days";
})(PlayTime || (PlayTime = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/refundpreference
 */
export var RefundPreference;
(function (RefundPreference) {
    RefundPreference[RefundPreference["Undeclared"] = 0] = "Undeclared";
    RefundPreference[RefundPreference["PreferAppleGrantsRefund"] = 1] = "PreferAppleGrantsRefund";
    RefundPreference[RefundPreference["PreferAppleDeclinesRefund"] = 2] = "PreferAppleDeclinesRefund";
    RefundPreference[RefundPreference["NoPreference"] = 3] = "NoPreference";
})(RefundPreference || (RefundPreference = {}));
/**
 * https://developer.apple.com/documentation/appstoreserverapi/userstatus
 */
export var UserStatus;
(function (UserStatus) {
    UserStatus[UserStatus["Undeclared"] = 0] = "Undeclared";
    UserStatus[UserStatus["AccountIsActive"] = 1] = "AccountIsActive";
    UserStatus[UserStatus["AccountIsSuspended"] = 2] = "AccountIsSuspended";
    UserStatus[UserStatus["AccountIsTerminated"] = 3] = "AccountIsTerminated";
    UserStatus[UserStatus["AccountHasLimitedAccess"] = 4] = "AccountHasLimitedAccess";
})(UserStatus || (UserStatus = {}));
